### PR TITLE
imporve:check the delete oparator in event expression

### PR DESCRIPTION
**Pull request by gebilaoxiong**
Fix: https://github.com/vuejs/vue/issues/4966

### CODE REVIEW

File: src/compiler/error-detector.js [hunk 1]
```diff
@@ -8,6 +8,8 @@ const prohibitedKeywordRE = new RegExp('\\b' + (
   'super,throw,while,yield,delete,export,import,return,switch,default,' +
   'extends,finally,continue,debugger,function,arguments'
 ).split(',').join('\\b|\\b') + '\\b')
+const deleteExpressionRE = /\bdelete\s*\([^\)]*\)/
```
Comment (HerringtonDarkholme) [inline src/compiler/error-detector.js]: What about `@input="delete"`?

Comment (gebilaoxiong) [inline src/compiler/error-detector.js]: This is the repair when the `delete()` method is called。
`@input="delete"` will be checked in` checkExpression` '

Comment (gebilaoxiong) [inline src/compiler/error-detector.js]: In the `with` statement call the `delete` method, always use the call is the global delete, which is a BUG need to compile at compile time

Comment (defcc) [inline src/compiler/error-detector.js]: There might be some other unary operators we need to check in theory, like `typeof`, `void` ?

**Comment by gebilaoxiong** [comment]
@defcc thanks : )

**Pull request merged**